### PR TITLE
Improve hooks documentation (spellcheck and position)

### PR DESCRIPTION
**Pull request by Iucapad**
This PR adds documentation for the recently added useSpellcheck hook, as well as the "fit" variant added for the position hook

### CODE REVIEW

File: content/developer/reference/frontend/hooks.rst [hunk 1]
```diff
@@ -282,3 +286,70 @@ API
...
+
+Location
+--------
+
+`@web/core/utils/hooks`
+
+Description
+-----------
+
+Activate the spellcheck state to an input or textarea on focus by a t-ref="spellcheck" in
```
Comment (fdardenne) [inline content/developer/reference/frontend/hooks.rst:303]: I would use more code quote in this section
```suggestion
Activate the spellcheck state to an input or textarea on focus by a `t-ref="spellcheck"` in
```

File: content/developer/reference/frontend/hooks.rst [hunk 2]
```diff
@@ -282,3 +286,70 @@ API
...
+.. list-table::
+   :widths: 20 20 60
+   :header-rows: 1
+
+   * - Option
+     - Type
+     - Description
+   * - `refName`
+     - string
+     - this is a `useRef reference <https://github.com/odoo/owl/blob/master/doc/reference/hooks.md#useref>`_ for the element that will be
```
Comment (fdardenne) [inline content/developer/reference/frontend/hooks.rst:354]: I don't know what it's worth, but there is an alias for the Owl GitHub now
```suggestion
     - this is a `useRef reference <{OWL_PATH}/doc/reference/hooks.md#useref>`_ for the element that will be
```

**Review by fdardenne — COMMENTED** [review]
It's nice to see more documentation!

**Review by fdardenne — APPROVED** [review]

**Comment by fdardenne** [comment]
@odoo/doc-review

File: content/developer/reference/frontend/hooks.rst [hunk 3]
```diff
@@ -282,3 +286,70 @@ API
...
+.. code-block:: xml
+
+    <t t-name="Comp" owl="1">
+      <input t-ref="spellcheck" type="text"/>
+      <textarea t-ref="custom"/>
+      <div t-ref="container">
+        <input type="text" spellcheck="false"/>
+        <div contenteditable="true"/>
+      </div>
+    </t>
```
Comment (AntoineVDV) [inline content/developer/reference/frontend/hooks.rst:336]: Use the `example` directive to highlight and clearly separate examples from the rest of the reference doc.
```suggestion
.. example::

   In the following example, the spellcheck will be enabled on the first input, the textarea and
   the div with `contenteditable="true"`.
   
   .. code-block:: javascript
   
       import { useSpellCheck } from "@web/core/utils/hooks";
       class Comp {
         setup() {
           this.simpleRef = useSpellCheck();
           this.customRef = useSpellCheck({ refName: "custom" });
           this.nodeRef = useSpellCheck({ refName: "container" });
         }
         static template = "Comp";
       }
   
   .. code-block:: xml
   
       <t t-name="Comp" owl="1">
         <input t-ref="spellcheck" type="text"/>
         <textarea t-ref="custom"/>
         <div t-ref="container">
           <input type="text" spellcheck="false"/>
           <div contenteditable="true"/>
         </div>
       </t>
```

**Review by AntoineVDV — APPROVED** [review]
@robodoo delegate+

**Comment by Iucapad** [comment]
@robodoo r+

**Pull request closed without merging**